### PR TITLE
Read all event trigger fix

### PR DESCRIPTION
**Pull request by rex-god**
- Only trigger read event on `read_all` _api_ if wallet not synced or last synced 1 day ago.
- When fetching data from the blockchain, save data immediately instead of envoke read event.

### CODE REVIEW

File: src/dal/wallet-last-sync-repo/wallet-last-sync.accessor.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { WalletLastSync, WalletLastSyncDocument } from './wallet-last-sync.schema';
+
+@Injectable()
+export class WalletLastSyncAccessor {
```
Comment (codinpsycho) [inline src/dal/wallet-last-sync-repo/wallet-last-sync.accessor.ts:7]: Why LastSync Accessor? That's a utility we are building on top of it. But in general, it's just a WalletAccessor

File: src/dal/wallet-last-sync-repo/wallet-last-sync.schema.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+import { Document } from 'mongoose';
+
+export type WalletLastSyncDocument = WalletLastSync & Document;
+
+@Schema({ timestamps: { createdAt: 'created_at', updatedAt: 'updated_at' } })
+export class WalletLastSync {
```
Comment (codinpsycho) [inline src/dal/wallet-last-sync-repo/wallet-last-sync.schema.ts:7]: WalletSchema

File: src/modules/helper/db-sync/nft-db-sync.service.ts [hunk 3]
```diff
@@ -72,41 +84,52 @@ export class NFtDbSyncService {
...
-      );
-      await this.nftInfoAccessor.updateManyNft(nftInfos);
+      await this.syncManyNftsFromBlockchain(event);
+      await this.walletDbSyncSvc.syncWallet(event.network, event.walletAddress);
     } catch (err) {
       console.error(err);
     }
   }
 
+  async syncManyNftsFromBlockchain(dto: syncManyNftDto): Promise<NftData[]> {
```
Comment (codinpsycho) [inline src/modules/helper/db-sync/nft-db-sync.service.ts:97]: This function doesn't sync from the blockchain, only updates DB with already synced NFT data

Comment (rex-god) [reply src/modules/helper/db-sync/nft-db-sync.service.ts:97]: no, it's updating data from blockchain too.

Comment (rex-god) [reply src/modules/helper/db-sync/nft-db-sync.service.ts:97]: Inside it's have `remoteDataFetcher.fetchAllNftDetails()`, which is reponsible to fetch data from blockchain.

Comment (codinpsycho) [reply src/modules/helper/db-sync/nft-db-sync.service.ts:97]: my  bad

**Pull request merged**